app-frontend/src/app/components/pages/project/layer/scenes/browse/index.js

```diff
@@ -1,11 +1,405 @@
+            icons: [
+                {
+                    icon: 'icon-eye',
+                    isActive: () => this.visibleScenes.has(scene.id)
+                }, {
+                    icon: 'icon-eye-off',
+                    isActive: () => !this.visibleScenes.has(scene.id)
+                }
+            ],
+            name: 'Preview',
+            tooltip: 'Show image boundaries on map',
+            callback: () => this.toggleVisibleScene(scene),
+            menu: false
+        }];
+
+        if (this.hasDownloadPermission(scene)) {
+            actions.unshift({
+                icon: 'icon-download',
+                name: 'Download',
+                tooltip: 'Download raw image data',
+                callback: () => this.modalService.open({
+                    component: 'rfSceneDownloadModal',
+                    resolve: {
+                        scene: () => scene
+                    }
+                }).result.catch(() => {}),
+                menu: false
+            });
+        } else {
+            actions.unshift({
+                icon: 'icon-item-lock',
+                name: 'This imagery requires additional access ' +
+                    `${this.currentRepository.service.permissionSource}`,
+                title: 'Download raw image data',
+                menu: false
+            });
+        }
+
+        return [scene.id, actions];
+    }
+
+    /**
+     * Convert a string in Leaflet bbox coordinate format ("swlng,swlat,nelng,nelat") to array
+     * @param {string} bboxString The bbox coordinate string to parse
+     * @return {array} lat/lon coordinates specifying bounding box corners ([[0,0], [1.0, 1.0]])
+     */
+    parseBBoxString(bboxString) {
+        let bbox = [];
+        if (bboxString && bboxString.length) {
+            let coordsStrings = bboxString.split(',');
+            let coords = _.map(coordsStrings, str => parseFloat(str));
+            // Leaflet expects nested coordinate arrays
+            bbox = [
+                [coords[1], coords[0]],
+                [coords[3], coords[2]]
+            ];
+        }
+        return bbox;
+    }
+
+    onRepositoryChange(bboxFetchFactory, repository) {
+        this.bboxFetchFactory = bboxFetchFactory;
+        this.currentRepository = repository;
+
+        this.sceneList = [];
+        this.sceneActions = new Map();
+        this.getMap().then((mapWrapper) => {
+            if (this.bboxCoords || this.queryParams && this.queryParams.bbox) {
+                this.fetchNextScenesForBbox = this.bboxFetchFactory(
+                    this.bboxCoords || this.queryParams.bbox
+                );
+                this.fetchNextScenes();
+            } else {
+                this.onViewChange(
+                    mapWrapper.map.getBounds(),
+                    mapWrapper.map.getCenter(),
+                    mapWrapper.map.getZoom()
+                );
+            }
+        });
+    }
+
+    onViewChange(newBounds, newCenter, zoom) {
```

Comment:
Hmm, I think it looks like duplicated logic, since there seems to be another same function that is supposed to do the similar thing in the same controller.

Reply:
er yup. didn't notice any issues since it was being overwritten by the second definition lol. If this function was called it would have errored out.